src/models/OuterControl.jl

```diff
@@ -41,8 +41,38 @@ function OuterControl(
     )
 end
 
-function OuterControl(; active_power, reactive_power, ext = Dict{String, Any}())
-    OuterControl(active_power, reactive_power, ext)
+function OuterControl(;
+    active_power,
+    reactive_power,
+    ext = Dict{String, Any}(),
+    states = nothing,
+    n_states = nothing,
+)
+    if isnothing(states)
+        @assert isnothing(n_states)
+        return OuterControl(active_power, reactive_power, ext)
+    end
+    @assert !isnothing(n_states)
+    return OuterControl(active_power, reactive_power, ext, states, n_states)
+end
+
+function IS.deserialize(::Type{T}, data::Dict) where {T <: OuterControl}
+    @debug "deserialize OuterControl" T data
+    vals = Dict{Symbol, Any}()
+    for (field_name, field_type) in zip(fieldnames(OuterControl), fieldtypes(OuterControl))
+        val = data[string(field_name)]
+        if field_name == :active_power
+            vals[field_name] = deserialize(T.parameters[1], val)
+        elseif field_name == :reactive_power
+            vals[field_name] = deserialize(T.parameters[2], val)
+        elseif field_name == :states
+            vals[field_name] = [Symbol(x) for x in val]
+        else
+            vals[field_name] = deserialize(field_type, val)
```

Comment:
```suggestion
        vals[field_name] = if field_name === :active_power
            deserialize(T.parameters[1], val)
        elseif field_name === :reactive_power
            deserialize(T.parameters[2], val)
        elseif field_name === :states
            [Symbol(x) for x in val]
        else
            deserialize(field_type, val)
```
similar cahnges apply to multiple places in the PR :)